src/configure/utilities/InputControlProvider.ts

```diff
@@ -0,0 +1,180 @@
+                    break;
+                case InputMode.Combo:
+                case InputMode.CheckBox:
+                case InputMode.RadioButtons:
+                    inputUxDescriptor = new InputUxDescriptor(input, inputUxDescriptorValue, ControlType.QuickPick);
+                    break;
+
+            }
+            if(inputUxDescriptor){
+                this._inputUxDescriptors.set(input.id, inputUxDescriptor);
+            }
+        }
+        this._setUxDescriptorsDataSourceInputs();   
+    }
+
+    private _setUxDescriptorsDataSourceInputs(): void {
+        this._uxInputs.forEach((inputDes) => {
+            if (!!inputDes.dataSourceId) {
+                var inputUxDescriptor = this._inputUxDescriptors.get(inputDes.id);
+                inputUxDescriptor.dataSource = DataSourceExpression.parse(inputDes.dataSourceId, this._pipelineTemplate.dataSources);
+
+                if (inputUxDescriptor.dataSource) {
+                    var dependentUxDesciptorArray = this._getInputDependencyArray(inputUxDescriptor, inputUxDescriptor.dataSource.getInputDependencyArray());
+                    if (dependentUxDesciptorArray) {
+                        inputUxDescriptor.dataSourceUxInputs.push(...dependentUxDesciptorArray);
+                    }
+                } else {
+                    throw new Error(`Data source {inputDes.dataSourceId} specified for input {inputDes.id} is not present in pipeline template {this._pipelineTemplate.id}`);
+                }
+            }
+        });
+    }
+
+    private _setupDefaultInputValue(inputUxDescriptor: InputUxDescriptor): void {
+        var inputDes = this._uxInputs.get(inputUxDescriptor.getInputUxDescriptorId());
+        if (!inputDes.defaultValue || !InputUxDescriptorUtility.doesExpressionContainsDependency(inputDes.defaultValue)) {
+            return;
+        }
+        var inputUxDescriptor = this._inputUxDescriptors.get(inputDes.id);
+        var dependentUxDesciptorArray = this._getInputDependencyArray(inputUxDescriptor, [inputDes.defaultValue], false);
+
+        var defaultValue = this._computeMustacheValue(inputDes.defaultValue, dependentUxDesciptorArray);
+        if (defaultValue !== inputUxDescriptor.getParameterValue()) {
+            inputUxDescriptor.setValue(defaultValue);
+        }
+    }
+
+    
+    private _computeMustacheValue(mustacheExpression: string, dependentUxDesciptorArray: InputUxDescriptor[]): string {
+
+        var dependentInputValues = this._getInputParameterValueIfAllSet(dependentUxDesciptorArray);
+        if (dependentUxDesciptorArray && dependentUxDesciptorArray.length > 0 && !dependentInputValues) {
+            return "";
+        } else {
+            return MustacheHelper.render(mustacheExpression, { inputs: dependentInputValues });
+        }
+    }
+
+    private _getInputUxDescriptorValue(inputDes: ExtendedInputDescriptor, inputValues: { [key:string]:any }) {
```

Comment:
should this not be named `_getInputValue` as this is returning value of the input?

Reply:
changed to getInputContolValue.